.pre-commit-config.yaml

```diff
@@ -81,50 +77,47 @@ repos:
           - .markdownlint.yaml
         # We would need to maintain format with changelog generator
         exclude: CHANGELOG.md
+
   - repo: https://github.com/k8s-at-home/sops-pre-commit
     rev: v2.1.1
     hooks:
       - id: forbid-secrets
+
+  - repo: https://github.com/rbubley/mirrors-prettier
+    rev: v3.6.2
+    hooks:
+      - id: prettier
+        args:
+          - --ignore-path
+          - .prettierignore
+          - --config
+          - .prettierrc.yaml
+
   - repo: https://github.com/gitleaks/gitleaks
     rev: v8.21.0
     hooks:
       - id: gitleaks
-  - repo: https://github.com/renovatebot/pre-commit-hooks
-    rev: 39.213.4
-    hooks:
-      - id: renovate-config-validator
-  - repo: https://github.com/codespell-project/codespell
-    rev: v2.4.1
-    hooks:
-      - id: codespell
-        files: ^.*\.(py|c|h|md|rst|yaml|toml|yml)$
-      - id: codespell
-        name: check commit message spelling
-        stages: [commit-msg]
-  - repo: https://github.com/python-poetry/poetry
-    rev: '2.1.1'
+
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.10.0
     hooks:
-      - id: poetry-check
-      - id: poetry-lock
-      - id: poetry-install
+      - id: python-check-mock-methods
+      - id: python-use-type-annotations
+      - id: rst-backticks
+      - id: rst-directive-colons
+      - id: rst-inline-touching-normal
+      - id: text-unicode-replacement-char
+
   - repo: local
     hooks:
       - id: basedpyright
         name: basedpyright
         description: "pyright fork with various type checking improvements, improved vscode support and pylance features built into the language server"
-        entry: poetry run basedpyright
+        entry: uv run basedpyright
```

Comment:
I've just found that this might not be the best idea to run basedpyright. This ignores any `--extra`  specification we might have in our venv. We often use the `--extra` feature to install or not install cuda dependencies. 
For instance, pytorch by default would install nvidia/cuda libraries. With `--extra cpu` we conventionally install it from the cpu-only index. If we then run `uv run basedpyright`, it will install the default dependencies from the default indices.
Maybe the best idea would be to just run `basedpyright`. This requires the `.venv` to be activated when running pre-commit, but I guess while developing your venv should always be activated anyway